rails/spec/requests/admin_user_request_spec.rb

```diff
@@ -0,0 +1,68 @@
+require 'rails_helper'
+
+RSpec.describe "SuperAdmin user request", type: :request do
+  describe "GET edit" do
+    context "with a super admin privileges" do
+      let(:super_admin) { FactoryBot.create(:user, role: 100, super_admin: true) }
+      let(:user) { FactoryBot.create(:user, email: "user@mail.co", username: "user_name", role: 2) }
```

Comment:
In case you were unaware, you can reference the roles by their symbols. So, rather than having to integer map `role: 2`, you can use `role: :admin`.